cogs/TFT.py

```diff
@@ -56,38 +57,37 @@ async def tftrank(self, ctx, region_code=None, *, summoner=None):
                 name="Incorrect command format!", value=info_msg)
         else:
             embed_msg = self.get_player_tft_rank(region_code, summoner)
-        await ctx.channel.send(embed=embed_msg)
+        await ctx.reply(embed=embed_msg)
 
     @commands.hybrid_command()
     @commands.check(checks.check_if_bot)
-    async def matchhistory(self, ctx, region_code=None, *, summoner=None):
+    async def matchhistory(self, ctx, region_code: str, *, summoner: str):
         """Prints the requested player's TFT match history (prev. 9 games) to Discord"""
+
         print(
             f"MATCHHISTORY / {str(summoner)} / {str(region_code)} / {ctx.author}")
 
-        if (region_code is None) or (summoner is None):
-            info_msg = "Command format should be: //matchhistory [region code] [summoner]\n Use //regions to see list " \
-                       "of correct region codes. "
-            embed_msg = discord.Embed(
+        improper_format_msg = "Command format should be: //matchhistory [region code] [summoner]\n Use //regions to see list " \
+                              "of correct region codes. "
+
+        error_embed_template = discord.Embed(
                 colour=discord.Colour.red()
             )
-            embed_msg.add_field(
-                name="Incorrect command format!", value=info_msg)
-            await ctx.channel.send(embed=embed_msg)
+
+        if (region_code is None) or (summoner is None):
+            error_embed_template.add_field(
+                name="Incorrect command format!", value=improper_format_msg)
+            await ctx.reply(embed=error_embed_template)
+            return
+        elif region_code.upper() not in decoder.region.keys():
+            error_embed_template.add_field(
+                name="Incorrect region code used!", value=improper_format_msg)
+            await ctx.reply(embed=error_embed_template)
+            return
         else:
-            # get region routing value OR send error message
-            try:
-                region_route = decoder.region[region_code.upper()]
-            except KeyError:
-                embed_msg = discord.Embed(
-                    color=discord.Colour.red()
-                )
-                msg = "Command format should be: //matchhistory [region code] [summoner] \n\
-                Use //regions to see list of correct region codes."
-                embed_msg.add_field(
-                    name="Incorrect region code used!", value=msg)
-                await ctx.channel.send(embed=embed_msg)
+            placeholder_msg = await ctx.reply("processing match history, please wait...")
```

Comment:
Could we make this more Neeko-fied? "Neeko goes on an adventure to find match history! Watch as Neeko uncovers the secrets of past battles with magical finesse! Stay close, summoner, for the enchanting revelations are about to unfold!" Courtesy of ChatGPT.

Reply:
Fixed in [b0443da](https://github.com/hschawe/NeekosHelp/pull/5/commits/b0443daec67184d96dfb23811656313c2d09ca13)